src/backend/impl/src/controllers/proposal_controller.rs

```diff
@@ -0,0 +1,189 @@
+use crate::{
+    repositories::{LogRepositoryImpl, ProposalRepositoryImpl, UserProfileRepositoryImpl},
+    services::{
+        AccessControlService, AccessControlServiceImpl, LogService, LogServiceImpl,
+        ProposalService, ProposalServiceImpl,
+    },
+};
+use backend_api::{ApiError, ApiResult, GetProposalsResponse};
+use candid::Principal;
+use ic_cdk::*;
+
+#[update]
+async fn sync_proposals() -> ApiResult<()> {
+    let calling_principal = caller();
+
+    ProposalController::default()
+        .sync_proposals(calling_principal)
+        .await
+        .into()
+}
+
+#[query]
+fn get_proposals() -> ApiResult<GetProposalsResponse> {
+    ProposalController::default().get_proposals().into()
+}
+
+pub(super) struct ProposalController<A: AccessControlService, L: LogService, P: ProposalService> {
+    access_control_service: A,
+    log_service: L,
+    proposal_service: P,
+}
+
+impl Default
+    for ProposalController<
+        AccessControlServiceImpl<UserProfileRepositoryImpl>,
+        LogServiceImpl<LogRepositoryImpl>,
+        ProposalServiceImpl<ProposalRepositoryImpl>,
+    >
+{
+    fn default() -> Self {
+        Self::new(
+            AccessControlServiceImpl::default(),
+            LogServiceImpl::default(),
+            ProposalServiceImpl::default(),
+        )
+    }
+}
+
+impl<A: AccessControlService, L: LogService, P: ProposalService> ProposalController<A, L, P> {
+    fn new(access_control_service: A, log_service: L, proposal_service: P) -> Self {
+        Self {
+            access_control_service,
+            log_service,
+            proposal_service,
+        }
+    }
+
+    async fn sync_proposals(&self, calling_principal: Principal) -> Result<(), ApiError> {
+        self.access_control_service
+            .assert_principal_is_admin(calling_principal)?;
+
+        self.proposal_service.fetch_and_save_nns_proposals().await
+    }
+
+    pub async fn sync_proposals_job(&self) {
```

Comment:
This one doesn't seem to be used yet, is there a plan to use it?

Reply:
It's used in the job in the init_controller:
https://github.com/CodeGov-org/codegov-website/blob/8044089d852c3c46e16ae1c56799dd47b9670788/src/backend/impl/src/controllers/init_controller.rs#L75